parquet-column/src/main/java/org/apache/parquet/column/values/delta/DeltaBinaryPackingValuesWriter.java

```diff
@@ -130,6 +135,21 @@ public long getBufferedSize() {
 
   @Override
   public void writeInteger(int v) {
+    /*
+     * isInt is needed. Just calling internalWrite might result in a different binary representation.
+     * Example: [Integer.MIN_VALUE, Integer.MAX_VALUE, Integer.MIN_VALUE, Integer.MAX_VALUE]
+     */
+    isInt = true;
```

Comment:
While we're at it, why not use int as the internal storage for the integer version? That way we don't use the extra memory (which could matter for staying in cache) and would avoid requiring the mask step just before flushing.